clients/src/test/java/org/apache/kafka/test/TestUtils.java

```diff
@@ -556,10 +558,20 @@ public static Set<TopicPartition> generateRandomTopicPartitions(int numTopic, in
      * @return The caught exception cause
      */
     public static <T extends Throwable> T assertFutureThrows(Future<?> future, Class<T> exceptionCauseClass) {
-        ExecutionException exception = assertThrows(ExecutionException.class, future::get);
-        assertInstanceOf(exceptionCauseClass, exception.getCause(),
-            "Unexpected exception cause " + exception.getCause());
-        return exceptionCauseClass.cast(exception.getCause());
```

Comment:
The goal of this PR, is to maintain the same semantics (the logic) as before, but make the future.get has time limit.
So, we should not change the assert logic here.
Maybe you should understand the original logic first, and then making other changes.
Some questions you need to know first is:
1. What will happen when future.get has no exception thrown?
2. What will happen when future.get throws ExecutionException?
3. What's exceptionCauseClass used for?
4. What's the return value?